### PR TITLE
[client] Add stateful userspace firewall and remove egress filters

### PR DESCRIPTION
**Pull request by lixmal**
## Describe your changes

- Add stateful firewall functionality for UDP/TCP/ICMP in userspace firewalll
- Removes all egress drop rules/filters, still needs refactoring so we don't add output rules to any chains/filters.
- on Linux, if the OUTPUT policy is DROP  then we don't do anything about it (no extra allow rules). This is up to the user, if they don't want anything leaving their machine they'll have to manage these rules explicitly.

### Checklist
- [x] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: client/firewall/uspfilter/conntrack/tcp.go [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+		timeout:       timeout,
+	}
+
+	go tracker.cleanupRoutine()
+	return tracker
+}
+
+// TrackOutbound processes an outbound TCP packet and updates connection state
+func (t *TCPTracker) TrackOutbound(srcIP net.IP, dstIP net.IP, srcPort uint16, dstPort uint16, flags uint8) {
+	t.mutex.Lock()
```
Comment (pappz) [inline client/firewall/uspfilter/conntrack/tcp.go]: We can apply the locks later, right before touch the map.

File: client/firewall/uspfilter/conntrack/tcp.go [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+	DestPort    uint16
+	State       TCPState
+	LastSeen    time.Time
+	established bool
+}
+
+// TCPTracker manages TCP connection states
+type TCPTracker struct {
+	connections   map[TCPConnKey]*TCPConnTrack
+	mutex         sync.RWMutex
```
Comment (pappz) [inline client/firewall/uspfilter/conntrack/tcp.go:70]: It is RWMutex but it has been used without RW feature

Comment (lixmal) [reply client/firewall/uspfilter/conntrack/tcp.go:70]: True. Looking at the locks we also modify conn under read locks

File: client/firewall/uspfilter/conntrack/tcp.go (hunk 1 above)
Comment (lixmal) [reply client/firewall/uspfilter/conntrack/tcp.go]: fixed

File: client/firewall/uspfilter/conntrack/tcp.go (hunk 2 above)
Comment (lixmal) [reply client/firewall/uspfilter/conntrack/tcp.go:70]: I've addressed the locking issues

**Review by mlsmaycon — APPROVED** [review]

**Pull request merged**